docs/RationaleSeq.md

```diff
@@ -106,3 +106,48 @@ information.
 - Initial value: this register is uninitialized. Using an uninitialized value
 results in undefined behavior. We will add an `initialValue` attribute if
 this proves insufficient.
+
+## The Register Operation
+
+`seq.reg` abstractly models a generic memory element that can be implemented
+as various physical memory elements. This attempts to capture most behaviours
+expressable by a `reg` in SystemVerilog.
+
+This is an abstract operation that should be further lowered down to either
+vendor-specific primtives or to the SV dialect.
+
+The register operation contains the following operands:
+
+- clock and its edge sensitivity: posedge, negedge, edge-triggered, or
+  unclocked. An unclocked register can be used to express a latch.
+- enable signals
+- input value
+- synchronous & asynchronus reset and reset values
+
+`seq.reg` has the following MLIR format:
+
+```mlir
+%q = seq.reg [posedge %clk | negedge %clk | edge %clk | unclocked],
+               %input, %enable,
+               %syncReset, %syncResetValue,
+               %asyncReset, %asyncResetValue
+               : $type(input)
+```
+
+### Rationale
+
+- Clock edges:
+  As a general-purpose register operation, this operation captures all possible
+  clock edge semantics that we will need in representing physical memory
+  elements.
+
+- Enable signal:
+  The enable signal is not optional. While the common-case is to set this to
+  vdd, this can be easily detected from analysis passes by checking if its
+  equal to `hw.constant 1`.
+
+- Non-optional syncReset and asyncReset:
+  Both synchronous reset and asynchronous resets are available as operands as
+  it is possible for flip flops to have both forms of reset. In the event that
+  these resets are not required, that can simply be tied to `hw.constant 0`,
+  and the resetValue operands can be populated with dummy values (eg: zeros)
```

Comment:
Would it be better to have resetValue optional so that we don't have to add constants just to have resetless regs?